plugins/pluginmanagers/lazy.nix

```diff
@@ -1,223 +1,187 @@
+        description = "List of plugins";
+      };
+    };
 
-              opts =
-                with helpers.nixvimTypes;
-                helpers.mkNullOrOption (maybeRaw (attrsOf anything)) ''
-                  opts should be a table (will be merged with parent specs),
-                  return a table (replaces parent specs) or should change a table.
-                  The table will be passed to the Plugin.config() function.
-                  Setting this value will imply Plugin.config()
-                '';
-            };
-          });
+  extraConfig =
+    cfg:
+    let
+      lazyPlugins = cfg.plugins;
 
-          listOfPlugins = types.listOf pluginType;
+      lazyPath =
+        let
+          processPlugin =
+            plugin:
+            [ (if lib.isDerivation plugin then plugin else plugin.pkg) ]
+            ++ lib.optionals ((plugin.dependencies or null) != null) (
+              builtins.concatMap processPlugin plugin.dependencies
+            );
         in
-        mkOption {
-          type = listOfPlugins;
-          default = [ ];
-          description = "List of plugins";
-        };
-    };
-  };
-
-  config = mkIf cfg.enable {
-    extraPlugins = [ pkgs.vimPlugins.lazy-nvim ];
-
-    extraPackages = [ cfg.gitPackage ];
-
-    extraConfigLua =
-      let
-        pluginToLua =
-          plugin:
-          let
-            keyExists = keyToCheck: attrSet: lib.elem keyToCheck (lib.attrNames attrSet);
-          in
-          if isDerivation plugin then
-            { dir = "${lazyPath}/${lib.getName plugin}"; }
-          else
-            {
-              "__unkeyed" = plugin.name;
-
-              inherit (plugin)
-                cmd
-                cond
-                config
-                dev
-                enabled
-                event
-                ft
-                init
-                keys
-                lazy
-                main
-                module
-                name
-                optional
-                opts
-                priority
-                submodules
-                ;
-
-              dependencies = helpers.ifNonNull' plugin.dependencies (
-                if isList plugin.dependencies then (pluginListToLua plugin.dependencies) else plugin.dependencies
-              );
-
-              dir =
-                if plugin ? dir && plugin.dir != null then plugin.dir else "${lazyPath}/${lib.getName plugin.pkg}";
-            };
+        pkgs.linkFarmFromDrvs "lazy-plugins" (builtins.concatMap processPlugin lazyPlugins);
 
-        pluginListToLua = map pluginToLua;
+      setupOptions =
+        let
+          pluginToSpec =
+            plugin:
+            if isDerivation plugin then
+              { dir = "${lazyPath}/${lib.getName plugin}"; }
+            else
+              plugin
+              // {
+                "__unkeyed" = plugin.name;
+                dir = if plugin ? dir && plugin.dir != null then plugin.dir else "${lazyPath}/${plugin.pkg.name}";
```

Comment:
> it doesn't work when using `plugin.pkg.name` in the if block above instead of `${lib.getName plugin}`

Interesting. Are you saying `pkgsLinkFarmFromDrvs` works or doesn't work?

`lib.getName` does do a bit of extra magic, for instance it can take either a derivation _or a string_. IDK if it does anything else different though...